confdgnmi/src/confd_gnmi_client.py

```diff
@@ -131,29 +137,39 @@ def read_subscribe_responses(responses, read_count=-1):
         log.info("==> read_count=%s", read_count)
         # example to cancel
         # responses.cancel()
-        for response in responses:
-            log.info("******* Subscription received response=%s read_count=%i",
-                     response, read_count)
-            print("subscribe - response read_count={}".format(read_count))
-            ConfDgNMIClient.print_notification(response.update)
-            if read_count > 0:
-                read_count -= 1
-                if read_count == 0:
-                    break
-        log.info("Canceling read")
+        num_read = 0
+        try:
+            for response in responses:
+                log.info("******* Subscription received response=%s read_count=%i",
+                         response, read_count)
+                print("subscribe - response read_count={}".format(read_count))
+                ConfDgNMIClient.print_notification(response.update)
+                num_read += 1
+                if read_count > 0:
+                    read_count -= 1
+                    if read_count == 0:
+                        break
+        except grpc.RpcError as e:
+            if isinstance(e, grpc.Call) and "EOF" in e.details():
+                log.warning("e.details=%s", e.details())
+            else:
+                raise e
```

Comment:
No need to `raise e`, just `raise` is enough, it also creates somewhat better stacktrace.